app/helpers/enterprises_helper.rb

```diff
@@ -99,4 +98,13 @@ def order_changes_allowed?
   def show_bought_items?
     order_changes_allowed? && current_order.finalised_line_items.present?
   end
+
+  def apply_stripe_filters_to(payment_methods)
+    stripe_enabled = Spree::Config.stripe_connect_enabled && Stripe.publishable_key
+    if stripe_enabled
+      payment_methods.reject!{ |p| p.type.ends_with?("StripeConnect") && p.preferred_enterprise_id == 0 }
+    else
+      payment_methods.reject!{ |p| p.type.ends_with?("StripeConnect") }
+    end
+  end
```

Comment:
I'm afraid we might be stuffing too much business logic into a helper.

First, helpers are meant to provide simple pure helper functions to the view. Secondly, as being modules they lack internal state, which makes us scatter such state throughout the methods with many arguments, and lack encapsulation.

IMHO things like this business logic shouldn't be in a helper but in single-purpose plain ruby objects with their own set of unit tests.

Reply:
Yep, thanks @sauloperez. I was struggling with that this afternoon. To split or not to split. I'll add it to my list. 🙂

Reply:
Resolved